### ISSUE TITLE
[reportArgumentType] For string , None can be assigned. But the pyright throws reportArgumentType error

### ISSUE THREAD

**Issue by rajk1729** [microsoft/pyright#9515]
**Describe the bug**
Getting this error for string assignment.
None is always assignable to string then why the following error for reportArgumentType.
For unknown|Any , we agree but for None, we can always assign None to string type.

```
error: Argument of type "Unknown | None" cannot be assigned to parameter "card_id" of type "str" in function "_get_card_details"
    Type "Unknown | None" is not assignable to type "str"
      **"None" is not assignable to "str" (reportArgumentType)**

```

**Code or Screenshots**

```
        address_id_list: list[str] = []
        billing_address_id = fetch_some_id() # this can be Null or any non blank string value
        address_id_list.append(billing_address_id)
```

**Comment by loing-cdut** [comment]
The "reportArgumentType" error occurs when using the mypy tool for static type checking in Python. This error indicates that you are trying to compare a type with another incompatible type or perform other operations.

In this case, you might be attempting to add a None type to a variable that requires a list of strings. To solve this problem, you can do some exception handling, such as:

```python
from typing import List

def fetch_some_id() -> str:
    # Implement the logic to fetch the ID here
    pass

address_id_list: List[str] = []
billing_address_id = fetch_some_id() # this can be Null or any non blank string value

if billing_address_id is not None:
    address_id_list.append(billing_address_id)
```

In this modified code, we use the `is not None` check to confirm that `billing_address_id` is not None, and then add it to the list. This avoids the "reportArgumentType" error.

**Comment by erictraut** [comment]
You haven't provided enough information to diagnose the issue. If you'd like me to investigate further, please provide a self-contained minimal code example that demonstrates the issue. The code you've provided above is incomplete and uses symbols that are not defined. 

Based on the error message, this is unlikely to be a bug in pyright, so I'm going to close the issue. If you still think that it's a real bug, please open a new issue with the information requested in the bug report template. If you have general questions about Python static typing, feel free to post a question in the discussion section.
